src/include/cart/crt_utils.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * (C) Copyright 2019-2021 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+/**
+ * Common functions to be shared among tests
+ */
+#ifndef __CART_UTILS_H__
+#define __CART_UTILS_H__
+#include <semaphore.h>
+#include <cart/api.h>
+#include "crt_internal.h"
+
+#define DBG_PRINT(x...)							\
+	do {								\
+		struct test_options *opts;				\
+		opts = crtu_get_opts();					\
+		D_INFO(x);						\
+		if (opts->is_server)					\
+			fprintf(stderr, "SRV [rank=%d pid=%d]\t",       \
+			opts->self_rank,				\
+			opts->mypid);					\
+		else							\
+			fprintf(stderr, "CLI [rank=%d pid=%d]\t",       \
+			opts->self_rank,				\
+			opts->mypid);					\
+		fprintf(stderr, x);					\
+	} while (0)
+
+struct test_options {
+	bool		is_initialized;
+	d_rank_t	self_rank;
+	int		mypid;
+	int		num_attach_retries;
+	bool		is_server;
+	bool		assert_on_error;
+	volatile int	shutdown;
```

Comment:
(style)  Missing a blank line after declarations
(style)  Use of volatile is usually wrong: see Documentation/volatile-considered-harmful.txt

Reply:
> (style) Use of volatile is usually wrong: see Documentation/volatile-considered-harmful.txt

We'll keep volatile here to prevent compiler optimization, which might cause the threads (main and progress) to read `shutdown` from the hardware cache instead of directly from memory.

> (style) Missing a blank line after declarations

We don't generally add blank lines in struct declarations, e.g., in [`src/include/cart/types.h`](https://github.com/daos-stack/daos/blob/32e81081a7ae75b369b8cacde5c97d485db16342/src/include/cart/types.h#L165).